### PR TITLE
Add support for dependency exclusion

### LINKED ISSUE #169: Allow explicit exclusion of dependencies from the jdeps module path
At the moment, Moditect simply uses `MavenProject.getArtifacts` to determine the module path to use when invoking `jdeps` to create `module-info`. While this is generally correct behavior, there are situations when it would be useful to be able to explicitly exclude some dependencies from the module path.

For example, when generating `module-info` for a module that shades one or more internal dependencies without relocation, not excluding the shaded dependencies from the module path leads to superfluous "split packages" warnings, as the same classes/packages exist in both artifacts. Excluding shaded dependency eliminates those warnings.

### PR DESCRIPTION
**Pull request by aseovic**
Fixes #169

### CODE REVIEW

**Review by gunnarmorling — APPROVED** [review]
LGTM; could you only add the new option to the description in the README, so folks know what it does? Sorry btw. for the long delay, somehow missed the notifications here.

**Comment by aseovic** [comment]
Sure, let me do that and I'll ping you back here.

File: maven-plugin/src/main/java/org/moditect/mavenplugin/add/AddModuleInfoMojo.java [hunk 1]
```diff
@@ -117,7 +127,13 @@ public void execute() throws MojoExecutionException, MojoFailureException {
     		getLog().debug("Mojo 'add-module-info' not executed on packaging type '"+project.getModel().getPackaging()+"'");
     		return;
     	}
-    	
+
+        if (exclusions != null) {
+            ArtifactFilter scopeFilter     = new ScopeArtifactFilter("compile+runtime");
```
Comment (aalmiray) [inline maven-plugin/src/main/java/org/moditect/mavenplugin/add/AddModuleInfoMojo.java:132]: Revisit this as technically `provided` scope could map to `requires static` instructions in the module descriptor.

**Comment by aalmiray** [comment]
I'll take care of the readme updates

**Pull request merged**